### PR TITLE
fix(Sunriver): /users call

### PR DESCRIPTION
**Pull request by plondon**
## Description
1. dont pass campaign metadata to /users call
2. pass newUser true only when applicable

## Change Type
Please enter one or more of the following: 
- Bug Fix

## Code Checklist
- [ ] Code compiles successfully (verified via `yarn start`)
- [ ] No lint issues exist (verified via `yarn lint`)
- [ ] New and existing unit tests pass (verified via `yarn test`)
- [ ] `README.md` and other documentation is updated as needed

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/identityVerification/sagas.js [hunk 1]
```diff
@@ -87,10 +87,10 @@ export default ({ api, coreSagas }) => {
       const userId = (yield select(
         selectors.core.kvStore.userCredentials.getUserId
       )).getOrElse('')
-      const userWithEmailExists = yield call(verifyIdentity)
-      if (userWithEmailExists) return
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/identityVerification/sagas.js]: Can someone explain why we need this?

File: packages/blockchain-wallet-v4-frontend/src/data/components/identityVerification/sagas.js [hunk 2]
```diff
@@ -87,10 +87,10 @@ export default ({ api, coreSagas }) => {
       const userId = (yield select(
```
[the remark this answers is not in the document]
Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/data/components/identityVerification/sagas.js]: We call with needsIdVerification for both a CREATED user and NONE

CREATED || ACTIVE: https://github.com/blockchain/blockchain-wallet-v4-frontend/blob/541ba65867af2b3ae6ef875821229f550dd8c701/packages/blockchain-wallet-v4-frontend/src/modals/Xlm/SunRiverWelcome/index.js#L119

NONE: https://github.com/blockchain/blockchain-wallet-v4-frontend/blob/541ba65867af2b3ae6ef875821229f550dd8c701/packages/blockchain-wallet-v4-frontend/src/modals/Xlm/SunRiverWelcome/index.js#L146

[the remark this answers is not in the document]
Comment (tony-blockchain) [reply packages/blockchain-wallet-v4-frontend/src/data/components/identityVerification/sagas.js]: it's purpose is to just create a user if there is none

[the remark this answers is not in the document]
Comment (tony-blockchain) [reply packages/blockchain-wallet-v4-frontend/src/data/components/identityVerification/sagas.js]: that means the CREATED || ACTIVE part is actually using the wrong action

File: packages/blockchain-wallet-v4-frontend/src/data/components/identityVerification/sagas.js [hunk 3]
```diff
@@ -54,7 +54,8 @@ export default ({ api, coreSagas }) => {
     coreSagas
   })
 
-  const registerUserCampaign = function*(newUser = false) {
+  const registerUserCampaign = function*(payload) {
```
Comment (tony-blockchain) [inline packages/blockchain-wallet-v4-frontend/src/data/components/identityVerification/sagas.js]: not exactly the same formatting that we use everywhere, it's usually `payload` property having everything inside

File: packages/blockchain-wallet-v4-frontend/src/modals/Xlm/SunRiverWelcome/index.js [hunk 4]
```diff
@@ -33,9 +33,28 @@ const Container = styled.div`
...
+  viewStellarWallet = () => {
+    this.props.modalActions.closeModal()
+    this.props.routerActions.push('/xlm/transactions')
+  }
+
   determineKycState (userState, kycState) {
+    // always register the campaign as quickly as possible if user has been created
     if (userState !== USER_ACTIVATION_STATES.NONE) {
-      this.props.registerSunRiverUser()
+      this.props.identityVerificationActions.registerUserCampaign()
```
Comment (tony-blockchain) [inline packages/blockchain-wallet-v4-frontend/src/modals/Xlm/SunRiverWelcome/index.js]: ideally we should move this to `componentDidUpdate` and `componentDidMount` as we don't want to fire actions on each render, can b done in a future update

**Pull request merged**